### PR TITLE
[liberasurecode] add new port

### PR DESCRIPTION
**Pull request by FredyVia**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

END OF NEW PORT CHECKLIST (delete this line) -->

### CODE REVIEW

File: ports/liberasurecode/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
+cmake_policy(SET CMP0054 NEW)
+
```
Comment (dg0yt) [inline ports/liberasurecode/portfile.cmake:2]:
```suggestion
```

File: ports/liberasurecode/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO openstack/liberasurecode
+    REF "${VERSION}"
+    SHA512 d5daa962324ef19fd195cfa842ec375d9dd5e62e3391b4a1cbf49a850b852b18cfc9be929ab18786d6b839139f6260d5cb4c88a0ba440c06b0e54e04ffb1bee1
+)
+
+set(ENV{CFLAGS} "-Wno-unused-variable -Wno-unused-but-set-variable")
+set(ENV{CXXFLAGS} "-Wno-unused-variable -Wno-unused-but-set-variable")
+
```
Comment (dg0yt) [inline ports/liberasurecode/portfile.cmake:12]:
```suggestion
```

File: ports/liberasurecode/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (dg0yt) [inline ports/liberasurecode/portfile.cmake:14]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/liberasurecode/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig()
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/erasurecode-1.pc" "@ERASURECODE_STATIC_LIBS@" "-lXorcode -lnullcode -lerasurecode_rs_vand")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/erasurecode-1.pc" "Requires: " "Requires: zlib")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/erasurecode-1.pc" " -lz" "")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/erasurecode-1.pc" "@ERASURECODE_STATIC_LIBS@" "-lXorcode -lnullcode -lerasurecode_rs_vand")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/erasurecode-1.pc" "Requires: " "Requires: zlib")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/erasurecode-1.pc" " -lz" "")
```
Comment (dg0yt) [inline ports/liberasurecode/portfile.cmake:24]: Maybe move to a patch, and upstream?

File: ports/liberasurecode/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "liberasurecode",
+  "version": "1.6.3",
+  "description": "Erasure Code API library written in C with pluggable Erasure Code backends. Mirror of code maintained at opendev.org.",
+  "homepage": "https://github.com/openstack/liberasurecode",
+  "license": "BSD-2-Clause",
+  "supports": "linux",
```
Comment (dg0yt) [inline ports/liberasurecode/vcpkg.json:7]: Why is it limited to Linux?

**Comment by Cheney-W** [comment]
@FredyVia Could you please solve the install failures? You could get the failure logs from https://dev.azure.com/vcpkg/public/_build/results?buildId=100008&view=artifacts&pathAsName=false&type=publishedArtifacts 
Thank you!

**Comment by Cheney-W** [comment]
[Cross-platform - liberasurecode is known to work on Linux (Fedora/Debian flavors), Solaris, BSD and Darwin/Mac OS X.](https://github.com/openstack/liberasurecode#:~:text=Cross%2Dplatform%20%2D%20liberasurecode%20is%20known%20to%20work%20on%20Linux%20(Fedora/Debian%20flavors)%2C%20Solaris%2C%20BSD%20and%20Darwin/Mac%20OS%20X.)

**Comment by FredyVia** [comment]
how could I run these check by my self(using workflow, or other ways, bot said: Commenter does not have sufficient privileges for PR 37063 in repo microsoft/vcpkg)

**Comment by FredyVia** [comment]
liberasurecode doesn't support windows, android

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**